libr/core/cmd_search.c

```diff
@@ -683,6 +681,13 @@ static int _cb_hit(RSearchKeyword *kw, void *user, ut64 addr) {
 	return true;
 }
 
+static int _cb_hit(RSearchKeyword *kw, void *user, ut64 addr) {
+	struct search_parameters *param = user;
+	const RSearch *search = param->core->search;
+	int klen = kw? kw->keyword_length + (search->mode == R_SEARCH_DELTAKEY): 0;
```

Comment:
so, it is valid to run this callback with kw as null? will be better to use R_NULLABLE in the signature definition. and i understand the reason for this case is when using search backends that don't use the concept of "keywords" like patterns and such, but maybe for those cases we can just holdl a fake instance in the stackframe and pass it, so you can keep using the kw->len instead of passing more arguments

Reply:
Check what I did and let me know.